src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpUpgradeProject.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using ProjectUtils = Microsoft.VisualStudio.IntegrationTest.Utilities.Common.ProjectUtils;
+
+namespace Roslyn.VisualStudio.IntegrationTests.Other
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpUpgradeProject : AbstractUpdateProjectTest
+    {
+        public CSharpUpgradeProject(VisualStudioInstanceFactory instanceFactory) : base(instanceFactory)
+        {
+        }
+
+        private void InvokeFixFromNewFile(ProjectUtils.Project project)
+        {
+            VisualStudio.SolutionExplorer.AddFile(project, "C.cs", @"
+class C
+{
+    int i = default;
+}", open: true);
+
+            VisualStudio.Editor.PlaceCaret("default");
+            VisualStudio.Editor.InvokeCodeActionList();
+            VisualStudio.Editor.Verify.CodeAction("Upgrade this project to C# language version '7.1'", applyFix: true);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUpgradeProject)]
+        public void CPSProject_GeneralPropertyGroupUpdated()
+        {
+            var project = new ProjectUtils.Project(ProjectName);
+
+            VisualStudio.SolutionExplorer.CreateSolution(SolutionName);
+            VisualStudio.SolutionExplorer.AddProject(project, WellKnownProjectTemplates.CSharpNetStandardClassLibrary, LanguageNames.CSharp);
+
+            InvokeFixFromNewFile(project);
+            VerifyPropertyOutsideConfiguration(GetProjectFileElement(project), "LangVersion", "7.1");
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CodeActionsUpgradeProject)]
+        public void LegacyProject_AllConfigurationsUpdated()
+        {
+            var project = new ProjectUtils.Project(ProjectName);
+
+            VisualStudio.SolutionExplorer.CreateSolution(SolutionName);
+            VisualStudio.SolutionExplorer.AddProject(project, WellKnownProjectTemplates.ClassLibrary, LanguageNames.CSharp);
+
+            InvokeFixFromNewFile(project);
+            VerifyPropertyInEachConfiguration(GetProjectFileElement(project), "LangVersion", "7.1");
+        }
```

Comment:
This needs to be fixed otherwise these tests will fail as soon as C# 8.0 is released with no obvious way how to fix them. @DustinCampbell do you have any suggestion on what to do with this?

Reply:
➡️ Fixed in 2b7dd403ea21b82c358044f46b4fb2073bd76d00